crates/python/src/qpu/client.rs

```diff
@@ -61,8 +203,15 @@ impl PyQcsClient {
                 Some(use_gateway) => client.with_use_gateway(use_gateway),
             };
 
-            Python::with_gil(|py| <_ as ToPython<Py<PyAny>>>::to_python(&Self(client), py))
-        })?
-        .extract()
+            Ok(Self(client))
+        })
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp, py: Python<'_>) -> PyObject {
+        match op {
+            CompareOp::Eq => (self == other).into_py(py),
+            CompareOp::Ne => (self != other).into_py(py),
+            _ => py.NotImplemented(),
```

Comment:
TIL that this (as you've written) is the right way to do it

https://docs.python.org/3.8/c-api/typeobj.html#c.PyTypeObject.tp_richcompare